src/libsystemd-network/dhcp6-option.c

```diff
@@ -529,6 +529,9 @@ int dhcp6_option_parse_ia(sd_dhcp6_client *client, DHCP6Option *iaoption, DHCP6I
                 if (len < DHCP6_OPTION_IA_NA_LEN)
                         return -ENOBUFS;
 
+                if (((const struct ia_na*) iaoption->data)->id != iaid)
```

Comment:
It would be nice to add a comment here mentioning RFC8415 I think, for future readers